contactaccessrights/contactaccessrights.php

```diff
@@ -190,3 +196,46 @@ function contactaccessrights_civicrm_pageRun($page) {
     CRM_Core_Resources::singleton()->addScriptFile($extName, 'js/dist/access-rights.min.js', 1010);
   }
 }
+
+/**
+ * Implementation of hook_addContactMenuActions to add the
+ * Contact Access menu group to the contact actions menu.
+ *
+ * @param ActionsMenu $menu
+ *
+ * @throws \Exception
+ */
+function contactaccessrights_addContactMenuActions(ActionsMenu $menu) {
+  $contactID = empty($_GET['cid']) ? '' : $_GET['cid'];
+  if (!$contactID) {
+    return;
+  }
+
+  $contactUserInfo = ContactHelper::getUserInformation($contactID);
+
+  if (empty($contactUserInfo['cmsId'])) {
+    return;
+  }
+
+  $userPermissions = CMSUserPermissionsFactory::create();
+
+  if (!$userPermissions->check($contactUserInfo, ['access CiviCRM'])) {
+    return;
+  }
+
+  $contactRightsService = new ContactRightsService();
+  $aclGroups = ContactHelper::getACLGroups($contactID);
+
+
+  $contactAccessActionGroup = new ContactAccessActionGroupHelper(
+    $contactUserInfo,
+    $contactRightsService,
+    $userPermissions,
+    $aclGroups
```

Comment:
I'm not sure if it's a good idea to pass the aclGroups to the ActionGroup like that, because, depending on how it's used it can lead to unexpected behaviors. For example, what happens if I pass an arrays of ACL groups for a contact B, while the information on `$contactUserInfo` is for contact A? Ideally, it would be better if we had some sort of ACL Service, which would expose a method to get the ACL groups of a contact. We would then inject that service here

Reply:
Ok, I can add a service for that. Ideally, where would be the best place or extension to put such a service?

Reply:
CRM_ContactActionsMenu_Service_ACLService?